ert3/evaluator/_evaluator.py

```diff
@@ -112,102 +168,29 @@ def _build_ee_config(
     input_records: MultiEnsembleRecord,
     dispatch_uri: str,
 ) -> Dict[str, Any]:
-    if ensemble.size != None:
-        ensemble_size = ensemble.size
-    else:
-        ensemble_size = input_records.ensemble_size
-
-    stage = stages_config.step_from_key(ensemble.forward_model.stage)
-    assert stage is not None
-    commands = (
-        stage.transportable_commands if isinstance(stage, ert3.config.Unix) else []
-    )
-    output_locations = [out.location for out in stage.output]
-    jobs = []
-
-    def command_location(name: str) -> FilePath:
-        return next(
-            (cmd.location for cmd in commands if cmd.name == name), pathlib.Path(name)
-        )
-
-    if isinstance(stage, ert3.config.Function):
-        jobs.append(
-            {
-                "name": stage.function.__name__,
-                "executable": cloudpickle.dumps(stage.function),
-                "output": output_locations[0],
-            }
-        )
-
-    if isinstance(stage, ert3.config.Unix):
-        for script in stage.script:
-            name, *args = script.split()
-            jobs.append(
-                {
-                    "name": name,
-                    "executable": command_location(name),
-                    "args": tuple(args),
-                }
-            )
+    ensemble_size = (
+        ensemble.size if ensemble.size != None else input_records.ensemble_size
+    ) or 0  # SMELL: ensemble_size is still optional since both option can be None
 
-    steps = [
-        {
-            "name": stage.name + "-only_step",
-            "inputs": [
-                {
-                    "record": input_.record,
-                    "location": input_.location,
-                    "mime": input_.mime,
-                    "is_executable": False,
-                }
-                for input_ in stage.input
-            ]
-            + [
-                {
-                    "record": cmd.name,
-                    "location": command_location(cmd.name),
-                    "mime": cmd.mime,
-                    "is_executable": True,
-                }
-                for cmd in commands
-            ],
-            "outputs": [
-                {
-                    "record": output.record,
-                    "location": output.location,
-                    "mime": output.mime,
-                }
-                for output in stage.output
-            ],
-            "jobs": jobs,
-            "type": "function" if isinstance(stage, ert3.config.Function) else "unix",
-        }
-    ]
+    step = stages_config.step_from_key(ensemble.forward_model.stage)
+    assert step is not None
 
-    ee_config: Dict[str, Any] = {
-        "steps": steps,
+    return {
+        "steps": _step(step),  # Why is steps a list of one dict object?
```

Comment:
Did you ever figure this out?

I think there are historical reasons, maybe @markusdregi could chime in.

Reply:
This has been discussed with Sandor and Markus. This is a feature that not yet implemented, but will in the future

Reply:
This is somewhat related to your first issue @sregales-TNO https://github.com/equinor/ert/issues/1553.

It is all connected... 🕸️